python/cudf/cudf/core/scalar.py

```diff
@@ -16,7 +17,50 @@
 )
 
 
-class Scalar(BinaryOperand):
+# Note that the metaclass below can easily be generalized for use with
+# other classes, if needed in the future. Simply replace the arguments
+# of the `__call__` method with `*args` and `**kwargs`. This will
+# result in additional overhead when constructing the cache key, as
+# unpacking *args and **kwargs is not cheap. See the discussion in GH
+# #11246 for more details.
+class CachedScalarInstanceMeta(type):
+    """
+    Metaclass for Scalar that caches `maxsize` instances.
+    """
+
+    def __new__(self, names, bases, attrs, maxsize=128):
```

Comment:
Canonically `__new__` accepts `cls` not `self` since it is implicitly a static method. I'm not sure if the conventions are different for metaclasses though... overriding `__new__` in a metaclass isn't something I really have experience with.
```suggestion
    def __new__(cls, names, bases, attrs, maxsize=128):
```

Reply:
Fix'd